docs/source/models/tool_calling.md

```diff
@@ -0,0 +1,286 @@
+# Tool Calling
+
+vLLM currently supports named function calling, as well as the `auto` and `none` options for the `tool_choice` field in the chat completion API. The `tool_choice` option `required` is **not yet supported** but on the roadmap.
+
+## Quickstart
+
+Start the server with tool calling enabled. This example uses Meta's Llama 3.1 8B model, so we need to use the llama3 tool calling chat template from the vLLM examples directory:
+
+```bash
+vllm serve meta-llama/Llama-3.1-8B-Instruct \
+    --enable-auto-tool-choice \
+    --tool-call-parser llama3_json \
+    --chat-template examples/tool_chat_template_llama3_json.jinja
+```
+
+Next make a request to extract structured data using function calling:
```

Comment:
"to extract structured data" doesn't quite seem right; that would be more appropriate e.g. for guided decoding / JSON mode. Maybe, "Next, make a request to the model that should result in it using the available tools"?